src/AutoRest.CSharp/Common/Input/CadlInputEnumTypeConverter.cs

```diff
@@ -74,6 +74,24 @@ public static InputEnumType CreateEnumType(ref Utf8JsonReader reader, string? id
             }
 
             valueType ??= InputPrimitiveType.Int32;
+            var currentType = InputPrimitiveType.Int32;
+            foreach (var value in allowedValues)
+            {
+                switch (value.Value)
+                {
+                    case Int32 i:
+                        if (currentType != InputPrimitiveType.Float32) currentType = InputPrimitiveType.Int32;
+                        break;
+                    case float f:
+                        currentType = InputPrimitiveType.Float32;
+                        break;
+                    case string:
+                        currentType = InputPrimitiveType.String;
+                        break;
+                }
+            }
```

Comment:
I get the logic here is to "go through all the allowed values to determine whether the underlying type of this should be int, float or string.
Should we also check if all the types are compatible? If one value is number, the other is string, this should not be allowed.